commit 5b919f833d9d60588d026ad82d17f17e8872c7a9
Author: Kees Cook <kees.cook@canonical.com>
Date:   Wed Jan 12 00:34:49 2011 -0800

    net: ax25: fix information leak to userland harder
    
    Commit fe10ae53384e48c51996941b7720ee16995cbcb7 adds a memset() to clear
    the structure being sent back to userspace, but accidentally used the
    wrong size.
    
    Reported-by: Brad Spengler <spender@grsecurity.net>
    Signed-off-by: Kees Cook <kees.cook@canonical.com>
    Cc: stable@kernel.org
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/ax25/af_ax25.c b/net/ax25/af_ax25.c
index bb86d29..6da5dae 100644
--- a/net/ax25/af_ax25.c
+++ b/net/ax25/af_ax25.c
@@ -1392,7 +1392,7 @@ static int ax25_getname(struct socket *sock, struct sockaddr *uaddr,
 	ax25_cb *ax25;
 	int err = 0;
 
-	memset(fsa, 0, sizeof(fsa));
+	memset(fsa, 0, sizeof(*fsa));
 	lock_sock(sk);
 	ax25 = ax25_sk(sk);
 
